aws-cpp-sdk-core/include/aws/core/utils/HashingUtils.h

```diff
@@ -107,7 +109,17 @@ namespace Aws
              */
             static ByteBuffer CalculateCRC32C(Aws::IOStream& stream);
 
-            static int HashString(const char* strToHash);
+            /**
+             * Hash a string to a 32-bit number.  Returning the native size_t
+             * would be better, but there seems to be a lot of code that expects
+             * this to return a signed int, so keeping it that way for now.
+             */
+            static inline int HashString(const char* strToHash) {
+                // take 0x00000FFFFFFFF000 from the 64-bit value
```

Comment:
Nice experiment

Reply:
I'd love to have fixed the rest of the SDK to use size_t but there are static const int declarations all over, and I'm not sure how all those files get generated or if I needed to power my way through it.